code/modules/power/power.dm

```diff
@@ -93,6 +93,11 @@
 		return
 	if(chan == -1)
 		chan = power_channel
+	var/avgrating = 1
+	for (var/obj/item/stock_parts/part in component_parts)
+		avgrating += part.rating
+	avgrating /= length(component_parts)
+	amount = amount ** avgrating;
```

Comment:
why exponential?? a machine that takes 500W now takes 62.5MW

Reply:
![image](https://user-images.githubusercontent.com/101573582/200801301-419c9cbc-097a-4097-8589-46798e7059af.png)

Reply:
Yeah it shouldn’t exponential like this

Reply:
> why exponential?? a machine that takes 500W now takes 62.5MW at T4

I like the concept of this PR but I too think the power consumption is a bit too much. Especially since people just go out of their way to upgrade every machine. They see once parts are available so it has to be made with most machines being upgraded in mind.

Reply:
i cut it in half

Reply:
also the whole point is to punish people for upgrading the clown's twinkie dispenser or some random machine